javascripts/mindmap.js

```diff
@@ -43,44 +43,57 @@ function getRandomId() {
 }
 
 /**
- * Recurse through the tree, generating IDs for all nodes missing them
+ * Normalize the raw YAML data into a tree structure expected by Mindmap.
```

Comment:
This is a little complex now, but other than practically testing, I'm not sure what to add. This project scores low on the Joel Test :D